### PR TITLE
Fix default viewers override

### PR DESCRIPTION
**Pull request by jtpio**
Troubleshooting https://github.com/jupyter/notebook/issues/6914

Notebook 7 should respect the `defaultViewers` settings overrides when opening a document.

Testing with Jupytext with the following:

- `pip install jupytext`
- `pip install git+https://github.com/mwouts/jupyterlab-jupytext.git@jupyterlab4`

### CODE REVIEW

**Comment by jtpio** [comment]
Still troubleshooting this.

**Comment by jtpio** [comment]
Looking better:

https://github.com/jupyter/notebook/assets/591645/6070024d-5a2f-42d5-bb41-be3d36e51d3a

Although not sure why there is no toolbar when opening the markdown file with the `Jupytext Notebook` factory. This also seems to be an issue with JupyterLab:


https://github.com/jupyter/notebook/assets/591645/293f0bae-59a6-4e96-ae1a-2cf144f73791

**Comment by jtpio** [comment]
cc @parmentelat for awareness, since this PR should already help in case you would like to try it out.

As noted above, it seems like there is an issue with the toolbar even with JupyterLab. Maybe this could be addressed as part of the JupyterLab 4 compatibility happening in https://github.com/mwouts/jupyterlab-jupytext/pull/3

File: packages/application-extension/src/index.ts [hunk 1]
```diff
@@ -195,10 +203,37 @@ const opener: JupyterFrontEndPlugin<void> = {
...
 
-        const file = decodeURIComponent(path);
-        const urlParams = new URLSearchParams(parsed.search);
-        const factory = urlParams.get('factory') ?? 'default';
         app.started.then(async () => {
+          const file = decodeURIComponent(path);
+          const urlParams = new URLSearchParams(parsed.search);
+          let defaultFactory = docRegistry.defaultWidgetFactory(path).name;
+
+          // Explicitely get the default viewers from the settings because
```
Comment (jtpio) [inline packages/application-extension/src/index.ts:211]: Maybe we can track that in an issue so it can be discussed.

**Review by andrii-i — APPROVED** [review]
Main needs to be merged in, otherwise looks good. Thank you @jtpio

**Pull request merged**